.github/workflows/test.yml

```diff
@@ -0,0 +1,56 @@
+name: Run tests on PR
+
+on:
```

Comment:
If you like, I can change to just `workflow_dispatch` to let users run GitHub Actions manually but conserve electricity and GitHub Actions credits if they do not run tests manually.